sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -207,15 +207,23 @@ private boolean validateSortDocument(List<Map<String, String>> sortDocument) {
                 } else {
                     accumulator = Sets.intersection(accumulator, childIds);
                 }
-
-                // TODO optimisation is to bail here if accumulator is empty
             }
 
             return accumulator;
         }
         if (node instanceof OrQueryNode) {
```

Comment:
This was a straight port from iOS version. https://github.com/cloudant/CloudantQueryObjc/blob/master/Pod/Classes/CDTQQueryExecutor.m#L240.  I can change it to an `else if` since I don't think functionally it will change anything but I wanted the Android and iOS versions to be as in synch as possible.  Let me know if you still rather it be set to an `else if`.